pkg/core/interop/runtime/witness.go

```diff
@@ -38,14 +38,14 @@ func checkScope(tx *transaction.Transaction, v vm.ScriptHashGetter, hash util.Ui
 			if c.Scopes&transaction.CalledByEntry != 0 {
 				callingScriptHash := v.GetCallingScriptHash()
 				entryScriptHash := v.GetEntryScriptHash()
-				if callingScriptHash == entryScriptHash {
+				if callingScriptHash.Equals(entryScriptHash) {
 					return true, nil
 				}
 			}
 			if c.Scopes&transaction.CustomContracts != 0 {
 				currentScriptHash := v.GetCurrentScriptHash()
 				for _, allowedContract := range c.AllowedContracts {
-					if allowedContract == currentScriptHash {
+					if allowedContract.Equals(currentScriptHash) {
```

Comment:
Not related to fees.

Reply:
That's unnecessary change, just removed this.